src/EncryptionOracle.sol

```diff
@@ -196,13 +156,50 @@ abstract contract EncryptionOracle is ThresholdNetwork, IEncryptionOracle, Ownab
         } catch Error(string memory reason) {
             revert OracleResultFailed(reason);
         } catch {
-            revert OracleResultFailed("Client does not support oracleResult() method");
+            revert OracleResultFailed(
+                "Client does not support oracleResult() method"
+            );
+        }
+    }
+
+    function estimateDeliverReencryption(
+        uint256 _requestId,
+        ReencryptedCipher calldata _cipher,
+        address callbackRecipient
+    ) external whenNotPaused onlyRelayer returns (bool) {
+        PendingRequest memory pr = pendingRequests[_requestId];
+        if (pr.client != address(0)) {
```

Comment:
Can you make a quick summary of why we have this if and what this function does different than the regular one, and why currently it's secure ?